giskard/llm/client/openai.py

```diff
@@ -147,7 +149,7 @@ def _completion(
 
         try:
             completion = self._client.chat.completions.create(
-                model=model,
+                model=self._model or model,
```

Comment:
Wrote also a comment below, this is bad because the Evaluators/Generators are also specifying the model, and here you just overwrite it.
I think you are right and it’s a problem with the current implementation. Let’s bind the model at the `LLMClient` level, passing it in init as you did here.
Let’s get rid of the `model` parameter when calling the `complete` method and let’s remove it from all generators/detectors. It will make things easier in the future.